clients/clients-integration-tests/src/test/java/org/apache/kafka/server/quota/CustomQuotaCallbackTest.java

```diff
@@ -69,10 +70,37 @@ public void testCustomQuotaCallbackWithControllerServer() throws InterruptedExce
                         && CustomQuotaCallback.COUNTERS.values().stream().allMatch(counter -> counter.get() > 0), 
                     "The CustomQuotaCallback not triggered in all controllers. "
             );
-            
+        
+        }
+    }
+
+    @ClusterTest(
+        types = {Type.CO_KRAFT, Type.KRAFT},
+        serverProperties = {
+            @ClusterConfigProperty(key = QuotaConfig.CLIENT_QUOTA_CALLBACK_CLASS_CONFIG, value = "org.apache.kafka.server.quota.CustomQuotaCallbackTest$MonitorableCustomQuotaCallback"),
         }
+    )
+    public void testMonitorableCustomQuotaCallbackWithCombinedMode(ClusterInstance cluster) {
+        assertEquals(2, MonitorableCustomQuotaCallback.metricNames.size());
+        MetricName metricName1 = MonitorableCustomQuotaCallback.metricNames.get(0);
+        MetricName metricName2 = MonitorableCustomQuotaCallback.metricNames.get(1);
+        if (metricName1.tags().get("role").equals(ProcessRole.ControllerRole.toString())) {
```

Comment:
If you use the actual value to predict the expected value, in the end you're not testing anything! Instead of that you can use `clusterInstance.type()` to identify the mode the test is using.